### DISCUSSION TITLE
USB-C: measure Battery status with 00100 get_battery_status

### DISCUSSION THREAD

**Discussion by maddhin** [arendst/Tasmota#22399] [Ideas]
Hi, I built a optical smart meter for my smart electric meter and have the problem that there is no power outlet available nearby to power the optical smart meter (ESP001)... 

So I power it off a power bank. Now the question is, how I can get the battery status of the power bank in order to track it and know when to recharge the bank. 

I read somewhere that the USB-C protokoll can call for a battery status. That would be super sleek but is there a way to realize this with Tasmota? What would be a realistic alternative way to get the battery status? 

Thanks a bunch for any comments! 
Martin

**Comment by sfromis** [comment]
Tasmota will only use USB for serial communication (USB CDC), and if you are not using ESP32-C3/S2/S3/C2/C6, the MCU chip does not even "see" USB, as there then will be a separate USB/serial adapter. Thus there is no way of doing what you envision, either way.

**Comment by maddhin** [comment]
That is a pity, so, basically there is no other way than dumping a perfectly good power bank for a 18650 battery shield and go for a voltage divider setup? 
Any other trick that might work (on any ESP). 

Cheers for the quick response!
